src/System.Text.Encoding/tests/Encoder/EncoderConvert2.cs

```diff
@@ -32,15 +32,34 @@ public override int GetBytes(char[] chars, int charIndex, int charCount, byte[]
     // Convert(System.Char[],System.Int32,System.Int32,System.Byte[],System.Int32,System.Int32,System.Boolean,System.Int32@,System.Int32@,System.Boolean@)
     public class EncoderConvert2
     {
-        #region Private Fields
         private const int c_SIZE_OF_ARRAY = 256;
         private readonly RandomDataGenerator _generator = new RandomDataGenerator();
-        #endregion
 
-        #region Positive Test Cases
-        // PosTest1: Call Convert to convert a arbitrary character array with UTF8 encoder
+        // Call Convert to convert a arbitrary character array with ASCII encoder
         [Fact]
-        public void PosTest1()
+        public void EncoderASCIIConvertRandomCharArray()
```

Comment:
>     [](start = 0, length = 4)

looks we have many methods which has exactly same code and differ in just one thing. like this method and the next 2 methods are same and differ in the encoder. can we try to to make it one theory method and pass the encoder as data to it?